packages/_shared/src/utils/entityHelpers.ts

```diff
@@ -199,14 +199,66 @@ export function getEntryTitle({
   return titleOrDefault(title, defaultTitle);
 }
 
-export function getEntryStatus(sys: Entry['sys']) {
+type AsyncPublishStatus = 'draft' | 'published' | 'changed';
+
+type FieldStatus = {
+  '*': Record<string, AsyncPublishStatus>;
+};
+
+export function getEntryStatus(
+  //TODO: remove union after fieldStatus is added to App SDK
+  sys: Entry['sys'] & { fieldStatus?: FieldStatus },
+  localeCodes?: string | string[]
+) {
   if (!sys || (sys.type !== 'Entry' && sys.type !== 'Asset')) {
     throw new TypeError('Invalid entity metadata object');
   }
+
   if (sys.deletedVersion) {
     return 'deleted';
   } else if (sys.archivedVersion) {
     return 'archived';
+  } else if (sys.fieldStatus) {
+    let status: AsyncPublishStatus = 'draft';
+
+    if (localeCodes) {
+      if (Array.isArray(localeCodes)) {
+        Object.entries(sys.fieldStatus['*']).forEach(([localeCode, fieldStatus]) => {
+          if (localeCodes.includes(localeCode)) {
+            if (fieldStatus === 'changed') {
+              status = fieldStatus;
+              return;
+            }
+            if (fieldStatus === 'published') {
+              status = fieldStatus;
+            }
+          }
+        });
+      } else {
+        Object.entries(sys.fieldStatus['*']).forEach(([localeCode, fieldStatus]) => {
+          if (localeCodes === localeCode) {
+            if (fieldStatus === 'changed') {
+              status = fieldStatus;
+              return;
+            }
+            if (fieldStatus === 'published') {
+              status = fieldStatus;
+            }
+          }
+        });
+      }
+    } else {
+      Object.values(sys.fieldStatus['*']).forEach((fieldStatus) => {
+        if (fieldStatus === 'changed') {
+          status = fieldStatus;
+          return;
+        }
+        if (fieldStatus === 'published') {
+          status = fieldStatus;
+        }
+      });
+    }
```

Comment:
What do you think about something like this?
Could make it a little bit tinier and easier to read?

```
const condition = (locale) => {
  if (Array.isArray(localeCodes)) {
    return localeCodes.includes(locale)
  }
  
  return localeCodes ? localeCodes === locale : true
}
    
Oject.entries(sys.fieldStatus['*']).forEach(([localeCode, fieldStatus]) => {
  if (condition(localeCode)) {
    if (fieldStatus === 'changed') {
      status = fieldStatus;
      return;
    }
    if (fieldStatus === 'published') {
      status = fieldStatus;
    }
  }
})